.github/bin/check_files.sh

```diff
@@ -12,6 +12,11 @@ function main {
     fi
   done
 
+  if ! jq -S .version ./test/expected_results.json; then
+    echo "🔥 jq cannot read file ./test/expected_results.json 🔥"
+    exit 1
+  fi
+
```

Comment:
If the problem is that jq wasn't installed in the container, I think the script should have this added rather this conditional.

```bash
function installed {
  cmd=$(command -v "${1}")

  [[ -n "${cmd}" ]] && [[ -f "${cmd}" ]]
  return ${?}
}

function die {
  >&2 echo "Fatal: ${@}"
  exit 1
}

# Check for all required dependencies
deps=(diff jq)
for dep in "${deps[@]}"; do
  installed "${dep}" || die "Missing '${dep}'"
done
```

This could be inserted  on line 27 before the call to `main`, but outside of the main function

Reply:
The problem is a bit more subtle than that. Normally when you're missing a dependency, the script will fail anyway in most cases when you try to use it. But in this case, the only usage of jq was this:
```
diff <(jq -S . file1.json) <(jq -S . file2.json);
```

I'm not an expert so I don't really understand why the exit code of `jq -S . file1.json` doesn't matter here, but the problem is that both commands `jq -S . file1.json` and `jq -S . file2.json` return exactly the same string "command not found: jq" which `diff` compares, sees they're identical and passes 🤦 

I'm not sure if explicitly checking if all dependencies are installed is the best solution here. It's more important to check that the commands inside `<()` do what they're supposed to, with or without js.

Reply:
yea, so in bash the `<(...)` evaluates the code inside of it, and it will capture `stdout`. But the error here is that jq wasn't installed, but we expect it to be, so we should still ensure that more explicitly.

Reply:
Sure, if you're giving me ready to use code so I don't have to figure it out on my own, why not :)